test/gpu/native/syncWarp.chpl

```diff
@@ -0,0 +1,34 @@
+use GPU;
+config const n=1000;
+on here.gpus[0] {
+  var Arr: [0..<n] int;
+  var Dummy: [0..<n] int;
+  var CorrectResult: [0..<n] bool;
+  @assertOnGpu
+  foreach i in Arr.domain {
+    const warpIndex = i/32;
+
+    if i%32==0 {
```

Comment:
I assume the idea is this `if` statement gets the first thread on the warp to do a bunch of work (assigning values into `Arr` that we'll read on your later `CorrectResult[i] = ...` line) and you have the other threads speed past it so if the `syncWarp()` doesn't occur then they'll have the wrong result when calculating `CorrectResult`?

Reply:
Yes that's the idea